### ISSUE TITLE
Progress Plugin Invalid Options

### ISSUE THREAD

**Issue by jimblue** [webpack/webpack#11371]
# Bug report

**What is the current behavior?**

Following this [Webpack documentation](https://webpack.js.org/plugins/progress-plugin/#root) and using the provide code giving the following error:

```
Progress Plugin Invalid Options
```

**If the current behavior is a bug, please provide the steps to reproduce.**

Adding the following code in the plugin section of **Webpack** config then run it.

```js
new webpack.ProgressPlugin({
  activeModules: false,
  entries: true,
  handler(percentage, message, ...args) {
     // e.g. Output each progress message directly to the console:
      console.info(percentage, message, ...args);
  },
  modules: true,
  modulesCount: 5000,
  profile: false,
  dependencies: true,
  dependenciesCount: 10000,
  percentBy: null
});
```

But I don't change the config of the plugin, it's perfectly working...

```js
const handler = (percentage, message, ...args) => {
    // e.g. Output each progress message directly to the console:
    console.info(percentage, message, ...args);
};

new webpack.ProgressPlugin(handler);
```

**What is the expected behavior?**

**Webpack** progress plugin should work with custom config like explain in the documentation.

Maybe it's a bug on the code or on the doc ?

**Other relevant information:**
webpack version:  4.44.1
Node.js version: 12.18.3
Operating System: MacOS Catalina 10.15.6

**Comment by vankop** [comment]
Please provide your config.

----
Note: you provided link to `webpack@5` documentation

**Comment by evilebottnawi** [comment]
I will reopen issue then you fill out the issue template

**Comment by vankop** [comment]
yep, it is bug in documentation there is not `dependencies`, `dependenciesCount`, `percentBy` options for webpack@4

**Comment by vankop** [comment]
@evilebottnawi please transfer to docs repo..
